[rpc][jit] support remote call in TorchScript

wanchaol opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#43046 [rpc][jit] support remote call in TorchScript**
* #43043 [rpc][jit] support rpc_sync in TorchScript
* #43042 [rpc] rename some functions

Differential Revision: [D23621108](https://our.internmc.facebook.com/intern/diff/D23621108)

wanchaol committed:
[WIP][rpc][jit] support remote call in TorchScript

[ghstack-poisoned]

wanchaol committed:
Update on "[WIP][rpc][jit] support remote call in TorchScript"




[ghstack-poisoned]

wanchaol committed:
Update on "[WIP][rpc][jit] support remote call in TorchScript"




[ghstack-poisoned]

wanchaol committed:
Update on "[WIP][rpc][jit] support remote call in TorchScript"




[ghstack-poisoned]

wanchaol committed:
Update on "[WIP][rpc][jit] support remote call in TorchScript"




[ghstack-poisoned]

wanchaol committed:
Update on "[WIP][rpc][jit] support remote call in TorchScript"




[ghstack-poisoned]

wanchaol committed:
Update on "[rpc][jit] support remote call in TorchScript"




[ghstack-poisoned]

wanchaol committed:
Update on "[rpc][jit] support remote call in TorchScript"


Differential Revision: [D23621108](https://our.internmc.facebook.com/intern/diff/D23621108)

[ghstack-poisoned]

mrshenli commented on torch/csrc/jit/frontend/ir_emitter.cpp:
```diff
@@ -3096,10 +3097,15 @@ struct to_ir {
+    } else {
+      output_type = returns[0].type();
     }
+    rpc_node_output->setType(output_type);
```
This looks a little different from the previous version. Prior to this change, if `rpc_op == prim::rpc_sync`, `rpc_node_output->setType` is not called. Is it true that, by default, its type is already `returns[0].type()`?

wanchaol replied:
good catch, no it's not automatically set I think, let me change it in previous PR

mrshenli reviewed (approved):
LGTM! Thanks for adding this!

wanchaol committed:
Update on "[rpc][jit] support remote call in TorchScript"


Differential Revision: [D23621108](https://our.internmc.facebook.com/intern/diff/D23621108)

[ghstack-poisoned]
